### ISSUE TITLE
SDL_Storage needs more documentation

### ISSUE THREAD

**Issue by NitroPlum** [libsdl-org/SDL#11784]
Could we update the SDL_Storage documentation to explain what TitleStorage and UserStorage are and maybe add a small snippet with an example usage? I was able to sort of guess what is going on but first-timers would be very lost.

**Comment by slouken** [comment]
I agree, I don't really understand how the API is intended to be used. @flibitijibibo, can you write an intro in the header like we did for the GPU API?

**Comment by NitroPlum** [comment]
This would be fantastic.

**Comment by NitroPlum** [comment]
I think I also found a bug but I will open that up in a new issue.

**Comment by flibitijibibo** [comment]
Working on this today - expect a draft before the new year.

**Comment by slouken** [comment]
Great, thanks!

**Comment by flibitijibibo** [comment]
The documentation changes are in - if there's anything else missing let me know, otherwise we should be all set!
